packages/repository-tests/src/crud/relations/acceptance/has-many-through.relation.acceptance.ts

```diff
@@ -205,9 +205,39 @@ export function hasManyThroughRelationAcceptance(
       expect(cartItems).have.length(0);
     });
 
-    //FIXME(Agnes): should be able to do deletion based on filters
+    it('deletes instances based on the filter', async () => {
+      const item1 = await customerRepo
+        .cartItems(existingCustomerId)
+        .create({description: 'group 1'});
+      await customerRepo
+        .cartItems(existingCustomerId)
+        .create({description: 'group 2'});
+
+      let links = await customerCartItemLinkRepo.find();
+      let cartItems = await cartItemRepo.find();
+      expect(links).have.length(2);
+      expect(cartItems).have.length(2);
+
+      await customerRepo
+        .cartItems(existingCustomerId)
+        .delete({description: 'group 3'});
```

Comment:
IIUC, here you are trying to delete a record that does not exist. That fact is not immediately clear from the code. Let's use a value that makes the intent more obvious.

```suggestion
        .delete({description: 'record-does-not-exist'});
```